### PR TITLE
Deptsortbycount; ArbitraryHtml option to wrap contents with p tag

### PR DESCRIPTION
**Pull request by amardesi**

### CODE REVIEW

File: app/jsx/pages/SearchPage.jsx [hunk 1]
```diff
@@ -360,72 +366,91 @@ class FacetFieldset extends React.Component {
     }
   }
```
Comment (amardesi) [inline app/jsx/pages/SearchPage.jsx]: Logic in code that follows has been drastically changed in order to better display checked facets.

File: app/jsx/pages/SearchPage.jsx [hunk 2]
```diff
@@ -542,11 +568,10 @@ class FacetForm extends React.Component {
       let filters = this.state.query.filters && this.state.query.filters[fieldName] ? this.state.query.filters[fieldName] : {}
       let facets = fieldset.facets
       return (
-        <FacetFieldset key={fieldName} index={"facetbox" + i} data={fieldset} query={filters} handler={this.changeFacet}
+        <FacetFieldset key={fieldName} index={"facetbox" + i} data={fieldset} query={filters}
+                       handler={this.changeFacet}
                        // Have first two open by default
-                       open={[0,1].includes(i)}
-                       modal={((["departments", "journals"].includes(fieldName)) &&
```
Comment (amardesi) [inline app/jsx/pages/SearchPage.jsx]: Logic moved inside FacetFieldset component where it's more relevant.

File: app/jsx/pages/SearchPage.jsx [hunk 3]
```diff
@@ -360,72 +366,91 @@ class FacetFieldset extends React.Component {
...
     /* Gather values of checked facets */
     let checkedHash = {}
     for (let filter of this.props.query.filters)
       checkedHash[filter.value] = true
-    let checked_count = Object.keys(checkedHash).length
 
     let checked = []
-    /* Can't use Generator function here since it's not supported by IE */
+    /* Collect checked facets into variable 'checked' (recursive)
+     * Can't use Generator function here since it's not supported by IE */
```
Comment (martinhaye) [inline app/jsx/pages/SearchPage.jsx]: Is that true even when we're using Babel? But in any case, this function is nice and clear, so it's fine.

**Comment by martinhaye** [comment]
Pushed to prd.

File: app/jsx/pages/SearchPage.jsx (hunk 3 above)
Comment (amardesi) [reply app/jsx/pages/SearchPage.jsx]: Ah I didn't take that into consideration.

**Pull request merged**